Mage.Sets/src/mage/cards/n/NettlingImp.java

```diff
@@ -0,0 +1,108 @@
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.SubtypePredicate;
+import mage.filter.predicate.permanent.ControlledFromStartOfControllerTurnPredicate;
+import mage.filter.predicate.permanent.ControllerPredicate;
+import mage.game.Game;
+import mage.target.common.TargetCreaturePermanent;
+import mage.watchers.common.AttackedThisTurnWatcher;
+
+/**
+ *
+ * @author MTGfan
+ */
+public class NettlingImp extends CardImpl {
+    
+    final static FilterCreaturePermanent filter = new FilterCreaturePermanent("non-Wall");
+
+    static {
+        filter.add(Predicates.not(new SubtypePredicate("Wall")));
+	filter.add(new ControlledFromStartOfControllerTurnPredicate());
+        filter.add(new ControllerPredicate(TargetController.ACTIVE));
+    }
+
+
+    public NettlingImp(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{B}");
+        
+        this.subtype.add("Imp");
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(1);
+
+        // {tap}: Choose target non-Wall creature the active player has controlled continuously since the beginning of the turn. That creature attacks this turn if able. If it doesn't, destroy it at the beginning of the next end step. Activate this ability only during an opponent's turn, before attackers are declared.
+        Ability ability = new ConditionalActivatedAbility(Zone.BATTLEFIELD, new AttacksIfAbleTargetEffect(Duration.EndOfTurn), new TapSourceCost(), new TurnCondition(), "{tap}: Choose target non-Wall creature the active player has controlled continuously since the beginning of the turn. That creature attacks this turn if able. If it doesn't, destroy it at the beginning of the next end step. Activate this ability only during an opponent's turn, before attackers are declared.");
+        ability.addEffect(new ConditionalOneShotEffect(new DestroyTargetAtBeginningOfNextEndStepEffect(), new InvertCondition(new TargetAttackedThisTurnCondition())));
+        ability.addTarget(new TargetCreaturePermanent(filter));
+        this.addAbility(ability, new AttackedThisTurnWatcher());
+        
+    }
+
+    public NettlingImp(final NettlingImp card) {
+        super(card);
+    }
+
+    @Override
+    public NettlingImp copy() {
+        return new NettlingImp(this);
+    }
+}
+
+class TurnCondition implements Condition {
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+	UUID activePlayerId = game.getActivePlayerId();
+        return activePlayerId != null && game.getPlayer(activePlayerId).hasOpponent(source.getControllerId(), game) && game.getPhase().getStep().getType().getIndex() < 5;
```

Comment:
Better get the active Player object and check that the Player is != null.
Here you can get a null pointer exceptions if the player left the game at the exactly that moment.